source/common/event/dispatcher_impl.cc

```diff
@@ -119,9 +119,9 @@ DispatcherImpl::createSslListener(Network::ConnectionHandler& conn_handler,
 }
 
 TimerPtr DispatcherImpl::createTimer(TimerCb cb) {
-  // TODO(htuch): Figure out why we can't set this assertion, see
-  // https://github.com/lyft/envoy/issues/1294.
-  // ASSERT(isThreadSafe());
+  // We don't ASSERT that createTimer() is called on the run thread because timers are used for
```

Comment:
If most of the cases are (1), then I think comments and a TODO for (2) are fine. If you just dump the above into a comment I think that would be in itself useful.

Maybe we can have a private `createTimerInternal()` that (1) use, and later switch to something like `postAfter()` if someone actually needs it.